File: db4oj/core/src/com/db4o/internal/marshall/SlotFormatCurrent.java
/* Copyright (C) 2008  Versant Inc.  http://www.db4o.com */

package com.db4o.internal.marshall;

import com.db4o.internal.*;
import com.db4o.typehandlers.*;


/**
 * @exclude
 */
public class SlotFormatCurrent extends SlotFormat {

    protected int handlerVersion() {
        return HandlerRegistry.HANDLER_VERSION;
    }

    public boolean isIndirectedWithinSlot(TypeHandler4 handler){
    	if (Handlers4.isUntyped(handler)) {
    		return false;
    	}
        return Handlers4.isVariableLength(handler) && Handlers4.isValueType(handler);
    }

}

File: db4o.net/Db4objects.Db4o/Db4objects.Db4o/Internal/Marshall/SlotFormatCurrent.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4objects.Db4o.Internal;
using Db4objects.Db4o.Internal.Marshall;
using Db4objects.Db4o.Typehandlers;

namespace Db4objects.Db4o.Internal.Marshall
{
	/// <exclude></exclude>
	public class SlotFormatCurrent : SlotFormat
	{
		protected override int HandlerVersion()
		{
			return HandlerRegistry.HandlerVersion;
		}

		public override bool IsIndirectedWithinSlot(ITypeHandler4 handler)
		{
			if (Handlers4.IsUntyped(handler))
			{
				return false;
			}
			return Handlers4.IsVariableLength(handler) && Handlers4.IsValueType(handler);
		}
	}
}
